src/functions/on-call/weekly-check/handler.ts

```diff
@@ -54,22 +53,20 @@ export const weeklyCheckHandler : ValidatedEventAPIGatewayProxyEvent<any> = asyn
     throw new Error("Next week not found");
   }
 
-  const yearFile = `${nextThursday.year}.json`;
-  const s3 = new S3();
-  const bucket = Config.get().onCall.bucketName;
-  const yearJson = (await S3Utils.loadJson<OnCallEntry[]>(s3, bucket, yearFile)) || [];
+  const dynamoDb = new DynamoDB.DocumentClient();
+  const year = nextThursday.year;
+  const week = nextWeek.week;
+  const person = nextWeek.user;
 
-  const selectedWeekIndex = yearJson.findIndex(entry => entry.Week == nextWeek.week);
-  if (selectedWeekIndex > -1) {
-    yearJson[selectedWeekIndex].Person = nextWeek.user;
-  } else {
-    yearJson.push({
-      Week: nextWeek.week,
-      Person: nextWeek.user
-    });
-  }
-
-  await S3Utils.saveJson(s3, bucket, yearFile, yearJson);
+  // Update or create a new record in DynamoDB
+  await dynamoDb.put({
```

Comment:
Again we really need a model to describe the structure of the data here- software-registry/ create-software might be a useful reference